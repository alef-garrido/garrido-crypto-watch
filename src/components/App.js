import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { loadCoins } from '../store/slices/coins-dux';
import Home from './Home';

function App() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(loadCoins());
  }, []);
  return (
    <main className="app--container">
      <div>
        <Home />
      </div>
    </main>
  );
}

export default App;
