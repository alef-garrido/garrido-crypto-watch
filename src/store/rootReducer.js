import { combineReducers } from 'redux';
import coins from './slices/coins-dux';

const rootReducer = combineReducers({
  coins,
});

export default rootReducer;
