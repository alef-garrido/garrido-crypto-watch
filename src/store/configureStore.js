import { configureStore } from '@reduxjs/toolkit';
import rootReducer from './rootReducer';
import apiCalls from './middleware/apiCalls';

const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(apiCalls),
});

export default store;
