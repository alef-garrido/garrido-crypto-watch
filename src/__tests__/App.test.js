import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store/configureStore';
import App from '../components/App';

it('renders correctly', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <App />
      </Provider>
      ,
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
