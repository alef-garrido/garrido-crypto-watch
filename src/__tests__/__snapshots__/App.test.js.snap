// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div>
    <div
      className="banner--container"
    >
      <p>
        Data provided by
        <a
          href="https://docs.coincap.io/"
        >
           CoinCap 2.0
        </a>
      </p>
    </div>
    <div
      className="app--container"
    >
      <main>
        <div
          className="head--display"
        >
          <header
            className="header--container"
          >
            <div
              className="header--title"
            >
              <div
                className="title--logo"
              >
                <img
                  alt=""
                  src="logo.png"
                />
              </div>
              <div>
                <h1>
                  Crypto
                  <br />
                  Watcher
                </h1>
                <p
                  className="header--subtitle"
                >
                  0 listed coins.
                </p>
              </div>
            </div>
            <form>
              <div
                className="header--search-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  />
                </svg>
              </div>
              <input
                onChange={[Function]}
                placeholder="Search"
                type="text"
                value=""
              />
            </form>
          </header>
        </div>
        <div
          className="list--display"
        >
          <div
            className="home--container"
          >
            <ul
              className="home--list-container"
            />
          </div>
        </div>
      </main>
    </div>
  </div>,
  ",",
]
`;
